components/tooltip/tooltip.js

```diff
@@ -650,6 +657,27 @@ class Tooltip extends RtlMixin(LitElement) {
 		return spaceAround;
 	}
 
+	async _determineIfTruncating() {
+		// if no resize has happened since truncation was previously calculated the result will not have changed
+		if (!this._resizeRunSinceTruncationCheck || !this.onlyShowIfTruncating) return;
+
+		const target = this._target;
+		const clone = target.cloneNode(true);
```

Comment:
I am curious what the behaviour would be for a HM component, or a component that does a lot of work when it is constructed or connected.  Presumably the clone automatically has all the internal DOM, but I am guessing it might do some HM request, but I am not sure.

Reply:
That could potentially be complicated - is there a case you have in mind where HM component content would potentially be truncated and show up in an overflow tooltip?

Reply:
`d2l-activity-name` and `d2l-hc-name` come to mind.  Maybe they would just want to bake the tooltip into them.  But then they would need to be focusable, ya?  Making things focusable that would not normally be focusable worries me though.

Reply:
If they did just baked it in, then the potentially truncated element would need to be focusable if they wanted the tooltip to show up on focus. If they just wanted to show tooltip on hover then I believe it should work out of the box.